app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -0,0 +1,193 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import Heatmap from "~/components/visualizations/heatmap/Heatmap";
+import { getAMRCounts } from "~/api/amr";
+
+import cs from "./amr_heatmap_vis.scss";
+
+const METRIC_ALLELES = "alleles";
+const METRIC_GENES = "genes";
+
+export default class AMRHeatmapVis extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      viewLevel: METRIC_GENES,
+      metric: "coverage",
+    };
+
+    this.heatmap = null;
+  }
+
+  componentDidMount() {
+    this.requestAMRCountsData(this.props.sampleIds);
+  }
+
+  componentDidUpdate() {
+    this.renderHeatmap();
+  }
+
+  async requestAMRCountsData(sampleIds) {
+    const rawSampleData = await getAMRCounts(sampleIds);
+    const samplesWithAMRCounts = rawSampleData.filter(
+      sampleData => sampleData.error === ""
+    );
+    const [samples, genes, alleles] = this.createLabels(samplesWithAMRCounts);
+    this.setState({
+      samplesWithAMRCounts,
+      samples,
```

Comment:
I would call these `sampleLabels`, `geneLabels` instead. Calling it `genes` implies that they're gene objects of some kind, instead of just the names.